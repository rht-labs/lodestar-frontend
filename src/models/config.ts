export interface Config {
  baseUrl: string;
  clientId: string;
  authBaseUrl: string;
  backendUrl: string;
}
export class Config {
  constructor({ baseUrl, clientId, authBaseUrl, backendUrl }: Config) {
    this.baseUrl = baseUrl;
    this.clientId = clientId;
    this.authBaseUrl = authBaseUrl;
    this.backendUrl = backendUrl;
  }

  baseUrl: string;
  clientId: string;
  authBaseUrl: string;
  backendUrl: string;

  static fromMap(map: { [key: string]: unknown }) {
    return new Config({
      baseUrl: map['baseUrl'] as string,
      clientId: map['clientId'] as string,
      authBaseUrl: map['authBaseUrl'] as string,
      backendUrl: map['backendUrl'] as string,
    });
  }
}
