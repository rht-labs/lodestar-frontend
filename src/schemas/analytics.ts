export interface AnalyticsEvent {
  category: string;
  action: string;
  label?: string;
  value?: number;
}
