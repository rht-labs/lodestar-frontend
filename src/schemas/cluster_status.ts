import { Subsystem } from './subsystem';
import { SystemMessage } from './system_message';
import faker from 'faker';

export enum HealthStatus {
  yellow = 'yellow',
  green = 'green',
  red = 'red',
}

export function getHumanReadableNameForHealthStatus(status: HealthStatus) {
  switch (status) {
    case HealthStatus.green:
      return 'Green';
    case HealthStatus.yellow:
      return 'Yellow';
    case HealthStatus.red:
      return 'Red';
  }
}

export function getColorForHealthStatus(status: HealthStatus) {
  switch (status) {
    case HealthStatus.green:
      return 'green';
    case HealthStatus.yellow:
      return '#EC7A08';
    case HealthStatus.red:
      return '#C9190B';
    default:
      return 'grey';
  }
}

export interface ClusterStatus {
  overall_status: HealthStatus;
  messages: SystemMessage[];
  subsystems: Subsystem[];
}

export abstract class ClusterStatus {
  static fromFake(staticData = false): ClusterStatus {
    return {
      overall_status: staticData
        ? HealthStatus.green
        : HealthStatus[HealthStatus[faker.random.number(2)]],
      messages: Array.apply(null, new Array(4)).map(x =>
        SystemMessage.fromFake(staticData)
      ) as SystemMessage[],
      subsystems: Array.apply(null, new Array(4)).map(x =>
        Subsystem.fromFake(staticData)
      ) as Subsystem[],
    };
  }
}
