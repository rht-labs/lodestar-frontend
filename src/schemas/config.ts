import { LogVerbosity } from '../utilities/logger/logger';

export interface BannerMessage {
  message: string;
  backgroundcolor: string;
}
export interface Config {
  baseUrl: string;
  clientId: string;
  authBaseUrl: string;
  backendUrl: string;
  disableLaunch: boolean;
  loggerType?: string;
  logLevel?: LogVerbosity;
  supportEmailAddress: string;
  bannerMessages?: BannerMessage[];
}

export abstract class Config {
  public static fromFake(): Config {
    return {
      baseUrl: 'http://localhost:3000',
      clientId: 'lodestar',
      authBaseUrl:
        'https://sso.example.com/auth/realms/omp/protocol/openid-connect',
      backendUrl: 'https://omp-backend.example.com',
      disableLaunch: false,
      supportEmailAddress: 'redhatsupport@redhat.com',
      bannerMessages: [
        {
          message: 'Hello message 1',
          backgroundcolor: '#FF0000',
        },
        {
          message: 'Hello message 2',
          backgroundcolor: '#FFFF00',
        },
        {
          message: 'Hello message 3',
          backgroundcolor: 'darkpurple',
        },
        {
          message: 'Hello message 4',
          backgroundcolor: 'brown',
        },
        {
          message: 'Hello message 5',
          backgroundcolor: 'green',
        },
        {
          message: 'Hello message 6',
          backgroundcolor: 'blue',
        },
        {
          message: 'Hello message 7',
          backgroundcolor: 'grey',
        },
        {
          message: 'Hello message 8',
          backgroundcolor: 'white',
        },
        {
          message: 'Hello message 9',
          backgroundcolor: 'indigo',
        },
      ],
    };
  }
}
