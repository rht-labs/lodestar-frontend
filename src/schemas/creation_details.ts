export interface CreationDetails {
  created_by_email: string;
  created_by_user: string;
  created_on: Date;
}
