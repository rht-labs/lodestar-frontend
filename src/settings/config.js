export const AppSettings = {
    baseUrl: process.env.REACT_APP_BASE_URL,
    clientId: process.env.REACT_APP_CLIENT_ID,
    redirectUri: process.env.REACT_APP_REDIRECT_URI,
    authBaseUrl: process.env.REACT_APP_AUTH_BASE_URL,
    backendUrl: process.env.REACT_APP_BACKEND_URI
}