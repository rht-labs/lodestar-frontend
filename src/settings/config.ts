export const AppSettings = {
    baseUrl: 'http://localhost:3000',
    clientId: 'open-management-portal',
    authBaseUrl: 'https://sso-omp-jasee.apps.s11.core.rht-labs.com/auth/realms/omp/protocol/openid-connect',
    backendUrl: 'http://localhost:3000/api'
}