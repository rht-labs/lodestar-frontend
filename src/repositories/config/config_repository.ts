import { Config } from '../../models/config';

export abstract class ConfigRepository {
  abstract async fetchConfig(): Promise<Config>;
}
