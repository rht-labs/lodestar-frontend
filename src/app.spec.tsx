import React from 'react';
import { App } from './app';
import { render } from '@testing-library/react';
import { Config } from './schemas/config';
describe('Root app', () => {
  test('matches snapshot', () => {
    expect(render(<App config={Config.fromFake()} />)).toMatchSnapshot();
  });
});
