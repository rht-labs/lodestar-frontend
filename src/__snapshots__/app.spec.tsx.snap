// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root app matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="landing-page"
      >
        <div
          class="pf-c-page"
        >
          <header
            class="pf-c-page__header"
            toolbar="[object Object]"
          >
            <div
              class="pf-c-page__header-brand"
            >
              <a
                class="pf-c-page__header-brand-link"
              >
                <span
                  class="logo-text"
                >
                  LodeStar
                </span>
              </a>
            </div>
          </header>
          <main
            class="pf-c-page__main"
            tabindex="-1"
          >
            <section
              class="pf-c-page__main-section"
              style="padding: 0px;"
            >
              <div
                style="height: 25px;"
              />
            </section>
            <div
              class="jumbotron-container"
            >
              <div
                class="cta"
              >
                <span>
                  LodeStar
                </span>
                <a
                  href="/app"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-primary"
                    data-cy="get_started_button"
                    data-ouia-component-id="0"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    Get Started
                  </button>
                </a>
              </div>
              <div
                class="background-image"
              />
              <div
                class="background-image-overlay"
              />
            </div>
            <section
              class="pf-c-page__main-section"
            />
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="landing-page"
    >
      <div
        class="pf-c-page"
      >
        <header
          class="pf-c-page__header"
          toolbar="[object Object]"
        >
          <div
            class="pf-c-page__header-brand"
          >
            <a
              class="pf-c-page__header-brand-link"
            >
              <span
                class="logo-text"
              >
                LodeStar
              </span>
            </a>
          </div>
        </header>
        <main
          class="pf-c-page__main"
          tabindex="-1"
        >
          <section
            class="pf-c-page__main-section"
            style="padding: 0px;"
          >
            <div
              style="height: 25px;"
            />
          </section>
          <div
            class="jumbotron-container"
          >
            <div
              class="cta"
            >
              <span>
                LodeStar
              </span>
              <a
                href="/app"
              >
                <button
                  aria-disabled="false"
                  class="pf-c-button pf-m-primary"
                  data-cy="get_started_button"
                  data-ouia-component-id="0"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Get Started
                </button>
              </a>
            </div>
            <div
              class="background-image"
            />
            <div
              class="background-image-overlay"
            />
          </div>
          <section
            class="pf-c-page__main-section"
          />
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
